core/drivers/clk/clk.c

```diff
@@ -38,9 +38,10 @@ struct clk *clk_alloc(const char *name, const struct clk_ops *ops,
 	return clk;
 }
 
-void clk_free(struct clk *clk)
+void clk_elt_free(struct clk *clk)
 {
-	free(clk);
+	if (clk)
+		free(clk_to_clk_elt(clk));
```

Comment:
To fix. `clk_to_clk_elt()` not yet defined at this stage of the patch series.
I'll fix:
```c
void clk_elt_free(struct clk *clk)
{
	free(clk);
}

Reply:
Actually this is a bad idea.
Clock are allocated by grapes with `clk_alloc()` and cannot be freed individually.
I'll revert this patch and update `clk_alloc()` / `clk_free()` inline description accordingly:
```patch
- * clk_free - Free a clock structure
+ * clk_free - Free clock array reference returned by clk_alloc()
  * 
- * @clk: Clock to be freed or NULL
+ * @clk: A reference returned by clk_alloc() or NULL